Modules/Sources/Yosemite/Tools/POS/POSCatalogPersistenceService.swift

```diff
@@ -68,6 +74,70 @@ final class POSCatalogPersistenceService: POSCatalogPersistenceServiceProtocol {
                       "\(variationImageCount) variation images, \(variationAttributeCount) variation attributes")
         }
     }
+
+    func persistIncrementalCatalogData(_ catalog: POSCatalog, siteID: Int64) async throws {
+        DDLogInfo("💾 Persisting incremental catalog with \(catalog.products.count) products and \(catalog.variations.count) variations")
+
+        try await grdbManager.databaseConnection.write { db in
+            let site = PersistedSite(id: siteID)
+            try site.insert(db, onConflict: .ignore)
```

Comment:
Should we ever insert a Site for an incremental persist? Perhaps this won't ever do anything but is good to have for safety

Reply:
Good catch - no, it's not necessary to insert a Site in an incremental sync as a full sync is a prerequisite. I removed it and updated test cases that depend on the Site insertion in https://github.com/woocommerce/woocommerce-ios/pull/16106/commits/cdd580d7cb758ae941e85719a70ff44e2b2c1c4c.